### DISCUSSION TITLE
Can cookbook version 5.0 be used in an esri 10.9.1 cloudformation template that references cookbook 3.8?

### DISCUSSION THREAD

**Discussion by timhaverland-noaa** [Esri/arcgis-cookbook#386] [Q&A] [answered]
Hi there,

I was having trouble building arcgis enterprise using a [10.9.1 esri cloudformation template](https://enterprise.arcgis.com/en/server/10.9.1/cloud/amazon/cf-enterprise-ha.htm) on Ubuntu 22.04. The stack build would fail on restarting the NFS service. After some research I realized that Ubuntu 22.04 uses a new configuration file setup, and that cookbook version 5.0 would be needed to properly set up NFS on Ubuntu 22.04.

The cloudformation template referenced above uses cookbook version 3.8. 

https://arcgisstore.s3.amazonaws.com/1091/cookbooks/arcgis-3.8.0-cookbooks.tar.gz

Do you think I could change the CookbooksUrl to 
https://arcgisstore.s3.amazonaws.com/1130/cookbooks/arcgis-5.0.0-cookbooks.tar.gz

And still be able to install arcgis enterprise version 10.9.1?

**Comment by cameronkroeker** [comment, marked as answer]
@timhaverland-noaa The 5.0.0 cookbooks do still contain the 10.9.1 templates so it might work, but without ever testing it I can't say for 100% certainty. I suggest first testing it to see if there are any quirks before fully rolling out in a production environment.

Also keep in mind that Ubuntu 22.04 is not in the supported OS list of 10.9.1:

https://enterprise.arcgis.com/en/system-requirements/10.9.1/linux/arcgis-server-system-requirements.htm#ESRI_SECTION1_B26370BE05834D038A68064E07B8036E

Thanks,
Cameron K.

**Reply by timhaverland-noaa** [reply]
Hi Cameron, thanks for the response. I will tweak the 10.9.1 cloudformation template to use the 5.0 cookbook and let you know how it goes. I did see that 22.04 is not in the supported list for 10.9.1, but this statement made me think that newer OS versions were supported: "Prior and future updates or service packs for these operating system versions are supported unless otherwise stated." But perhaps that means updates/service packs only within Ubuntu 18.xx 20.xx?

**Reply by cameronkroeker** [reply]
Right I believe it means minor version updates/service packs for 18.xx and 20.xx (whereas 18, 20, 22 are major versions that require a full OS upgrade). It is also possible that 10.9.1 works on 22.xx but since 22.xx was likely released after 10.9.1 it was never certified for it.

**Reply by timhaverland-noaa** [reply]
Got it, thank you.
